apps/ecommerce/lambda/src/routers/shopifyRouter.spec.ts

```diff
@@ -0,0 +1,118 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import { mockResourceLink } from '../mocks/resourceLink.mock';
+import Sinon from 'sinon';
+import Client, { CheckoutResource, CollectionResource, ImageResource, Product, ProductResource, Shop, ShopResource } from 'shopify-buy';
+
+const sandbox = Sinon.createSandbox();
+chai.use(chaiHttp);
+chai.should();
+
+const shopifyClientStub = {
+    product: {
+        fetch: () => Sinon.promise((resolve) => {
+            return resolve({
+                title: 'Some Product title',
+                description: 'Some product description',
+                images: [{ url: 'some-product-url' }]
+            } as Product)
+        }),
+        fetchAll: () => Sinon.promise(Sinon.spy()),
```

Comment:
Seems like a lot of overhead for a Sinon stub, but I kept running into errors unless I implemented spies for every method contained in the SDK. Open to any feedback on how to do this better!

Reply:
yeah that's weird but I think what you've got here makes sense given that's the case!